src/coverlet.core/Instrumentation/Instrumenter.cs

```diff
@@ -177,18 +177,25 @@ private void InstrumentModule()
 
                     foreach (TypeDefinition type in types)
                     {
-                        var actualType = type.DeclaringType ?? type;
-                        if (!actualType.CustomAttributes.Any(IsExcludeAttribute)
+                        TypeDefinition actualType = type.DeclaringType ?? type;
+                        if (
+                            !actualType.CustomAttributes.Any(IsExcludeAttribute)
+
                             // Instrumenting Interlocked which is used for recording hits would cause an infinite loop.
                             && (!_isCoreLibrary || actualType.FullName != "System.Threading.Interlocked")
-                            && !_instrumentationHelper.IsTypeExcluded(_module, actualType.FullName, _excludeFilters)
+
+                            && (
+                                 !_instrumentationHelper.IsTypeExcluded(_module, actualType.FullName, _excludeFilters) &&
+                                 // Check exclusion for nested types
+                                 (type.IsNested ? !_instrumentationHelper.IsTypeExcluded(_module, type.FullName, _excludeFilters) : true)
```

Comment:
@SteveGilham this is the core update, nested types weren't checked for exclusion filtering.

Reply:
@tonerdo do you remember the reason?

Reply:
Wouldn't it be simpler just to write the check as an iteration up the declaration chain like
```
    var excludedByName = false;
    for (var checkType = type; checkType != null && !excludedByName; checkType = checkType.DeclaringType) 
        excludedByName = _instrumentationHelper.IsTypeExcluded(_module, checkType.FullName, _excludeFilters)
```

Reply:
You mean if current type or one of his parent type are excluded exclude it. Is this the correct semantic?I mean suppose deep nesting A->B->C and I want to skip only B, if I understood well in this case we'll skip also C, right?

Reply:
Actually at the moment the behavior is what you expect with only the first direct parent, so the semantic should be if I'm out also all my children and children of my children will be out. I'll give it a try with your idea Steve thanks!

Reply:
Of course, as `type.DeclaringType.FullName` is always a leading substring of `type.FullName`, simply taking a string `"*MatchEndOfClassName"` to regex `".*MatchEndOfClassName\/?" (as Cecil indicates nesting by `/``), the match could be simply done without looking for declaring types or being nested.

Iteration would still be required to pick up things like exclusions by attribute, though